### ISSUE TITLE
Warning for bitwise XOR with float values for PHP<=8.0

### ISSUE THREAD

**Issue by JacekAndrzejewski** [kalessil/phpinspectionsea#1788]
Description:
--

Currently it's very easy to miss usage of a bitwise XOR instead of a power operator. Easily can happen when copying formulas from excel, wolfram alpha or most algebra systems.
As floats are first cast to integers and then XORed operation like that is a result of a mistake or a shortcut for `(int)$a ^ (int)$b`. Either way it's hard to spot and it's not obvious how it behaves.

I think there is no good way to handle this for integers with pure static analysis. Something like detection if user pastes in the code from outside IDE and pasted code has the ^ sign could help, but I don't think that would really be helpful.

As such I would settle for a warning when there is a bitwise XOR between something other than two integers.

This is a deprecation in PHP 8.1, but for lower versions it's still a pretty hard to find bug.

Example
--
```PHP
$x = 11.54;
$power = 0.6;

$approximationY = $x ^ $power + 12.5; //WHOOPS, XOR instead of ** !!!
//it's identical in behaviour to
$approximationY = ((int)$x ^ (int)$power) + 12.5;
//in PHP 8.1 gives this notice
//Deprecated: Implicit conversion from float 11.54 to int loses precision
```
